pkgs/development/libraries/libunistring/default.nix

```diff
@@ -1,42 +1,56 @@
-{ fetchurl, lib, stdenv, libiconv, updateAutotoolsGnuConfigScriptsHook }:
+{
+  fetchurl,
+  lib,
+  stdenv,
+  libiconv,
+  updateAutotoolsGnuConfigScriptsHook,
+  darwin
+}:
 
 # Note: this package is used for bootstrapping fetchurl, and thus
 # cannot use fetchpatch! All mutable patches (generated by GitHub or
 # cgit) that are needed here should be included directly in Nixpkgs as
 # files.
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "libunistring";
-  version = "1.1";
+  version = "1.2";
 
   src = fetchurl {
-    url = "mirror://gnu/libunistring/${pname}-${version}.tar.gz";
-    sha256 = "sha256-oiUr7uyDCsREufaNazitiD2xmRnbNbUiIs+CfDhb22o=";
+    url = "mirror://gnu/libunistring/libunistring-${finalAttrs.version}.tar.gz";
+    hash = "sha256-/W1WYvpwZIfEg0mnWLV7wUnOlOxsMGJOyf3Ec86rvI4=";
   };
 
-  outputs = [ "out" "dev" "info" "doc" ];
+  outputs = [
+    "out"
+    "dev"
+    "info"
+    "doc"
+  ];
 
   strictDeps = true;
   propagatedBuildInputs = lib.optional (!stdenv.isLinux) libiconv;
+  buildInputs = lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk_11_0.frameworks.CoreServices
+  ];
```

Comment:
This breaks the build on x86_64-darwin. SDKs can’t be mixed like that. Unless libunistring _needs_ the 11.0 SDK, it should use an unversioned reference. Otherwise, it should use an unversioned reference and `overrideSDK` to change the SDK version.

Reply:
Good to know I thought that 11.0 SDK was preferred in all cases. Thanks a lot for the detailed explanation.

Reply:
Dealing with SDK versions should hopefully get better in the future, but right now it can be a pain. 😕

As a rule of thumb, go with the unversioned references unless the SDK is missing symbols from a newer one. If it’s missing symbols, hope use the 11.0 SDK and hope that’s good enough. There are a few things that require the 12.0 SDK at least, which can’t be built right now.